src/backend/distributed/metadata/node_metadata.c

```diff
@@ -1556,9 +1557,11 @@ AddNodeMetadata(char *nodeName, int32 nodePort,
  * SetWorkerColumn function sets the column with the specified index
  * (see pg_dist_node.h) on the worker in pg_dist_node.
  * It returns the new worker node after the modification.
+ * It also sends commands for the same update on the other metadata nodes,
+ * unless localOnly is true.
  */
-static WorkerNode *
-SetWorkerColumn(WorkerNode *workerNode, int columnIndex, Datum value)
+WorkerNode *
```

Comment:
I think a better API would look like:
```
/* which does local only */
SetWorkerColumnLocalOnly(workerNode, index, value)

/* by default, we  do this on all MX nodes */
SetWorkerColumn(workerNode, index, value)
```

We need to write explaining comments.